Compare to brain segmentation in nilearn_prototype.py workflow

Leengit opened the pull request:
ENH: Add segmentation files generated with `transformSegmentationsToTemplate.sh` and `combineSegmentations.py`
ENH: Write `find_local_maxima()`
ENH: Show use of user-supplied segmentation to annotate maxima

For example, in predicting voxel data based upon whether `demo_gender_id_v2` (from `gender-identity-sexual-health/gish_p_gi.csv`) equals 1, we get:
```text
Found local maximum -log_10 p(t-stat)=2.568679663117875 at (44, 66, 84 = 139-55) in region CST_right (15)
Found local maximum -log_10 p(t-stat)=1.4908409049457598 at (64, 72, 68 = 139-71) near region FPT_right (17)
Found local maximum -log_10 p(t-stat)=0.8190578464901322 at (58, 39, 50 = 139-89) near region MCP (26)
Found local maximum -log_10 p(t-stat)=0.6572261126411297 at (83, 39, 45 = 139-94) in region ICP_left (20)
Found local maximum -log_10 p(t-stat)=0.37555506476341377 at (84, 88, 76 = 139-63) in region IFO_left (22)
Found local maximum -log_10 p(t-stat)=0.33044764625254935 at (54, 59, 87 = 139-52) in region AF_right (1)
Found local maximum -log_10 p(t-stat)=0.3142121526507991 at (57, 45, 101 = 139-38) in region ST_PAR_right (48)
Found local maximum -log_10 p(t-stat)=0.19075312610054695 at (56, 55, 73 = 139-66) in region OR_right (30)
Found local maximum -log_10 p(t-stat)=0.19007990556284868 at (86, 56, 80 = 139-59) in region ST_PAR_left (47)
```
Modulo the possibility of hallucinations, the above results look promising:
* When I Google for "right Corticospinal Tract gender" the AI responds "In general, males tend to have greater fractional anisotropy (FA) in the right corticospinal tract compared to females."  
* For "right Frontoparietal-Temporal gender", Google's AI reports "Research suggests that there are gender differences in the right Frontoparietal-Temporal areas of the brain, specifically in how these areas function and connect. Studies have shown that males may exhibit stronger rightward lateralization of brain connectivity in these regions, while females may show greater leftward lateralization."
* For "right Middle cerebellar peduncle gender" the result is less encouraging: "While there isn't a direct, well-established gender-specific difference in the anatomy or function of the right middle cerebellar peduncle (MCP) itself, research suggests potential interactions between sex and MCP white matter properties, particularly in relation to Fragile X-associated tremor/ataxia syndrome (FXTAS) and internalizing symptoms."

Note that the third coordinates in the workflow and the third coordinates in 3D Slicer add to 139 (where 139 is relevant because voxel coordinate values run from 0 to 139).  So, e.g., with 3D Slicer use third coordinate equal to 55, to examine the first maximum above, but use third coordinate equal to 84 if looking through data structures within the notebook.

Leengit commented:
Also note that some maxima say "in region" and some say "near region".  The workflow reports "near region" if the voxel at the intensity maximum is considered `Background (-1)` but it finds that a plurality of the non-background neighboring voxels are from a non-background brain segment.

ebrahimebrahim reviewed (commented):
This is great, and I'm so happy to see that the results of find_local_maxima look promising!

Does this mean we are ready to present what we have and get some feedback from Martin? Do you feel like you are ready to pull together the materials to present it in an informal discussion setting?

---

For the PR, let's not include the data files, but instead add a note to the readme that these data files are needed for the nilearn_prototype notebook and include a link to download them. You can host them in a public or kitware-only google drive folder, or on data.kitware.com.

Leengit commented:
I fixed a bug that still had a `-1` where a `0` should have been used to index `Background`.  I added some mathematical description to `nilearn_prototype.py` and a few more details about the output of the pipeline.

ebrahimebrahim reviewed (approved):
The high level description looks like exactly what I was hoping for, thanks!
